src/ResourceManager/IotHub/Commands.IotHub/IotHub/AddAzureRmIotHubKey.cs

```diff
@@ -50,15 +50,13 @@ public class AddAzureRmIotHubKey : IotHubBaseCmdlet
         public string KeyName { get; set; }
 
         [Parameter(
-            Mandatory = true,
+            Mandatory = false,
```

Comment:
Neither parameter is mandatory?  If one or the other is required, you should set up parameter sets to require specifying one